Sage/carbon
# fix(dialog, dialog-full-screen, sidebar, form): ensure focused elements are displayed on screen FE-5236

fix #5278

### Proposed behaviour


Adds `useIsStickyFooterForm` hook to `Dialog`, `DialogFullScreen` and `Sidebar` components to
detect when a `Form` with `stickyFooter` is a child. If one is detected it unsets the overflow and
uses the `Form`'s overflow to support scrolling. There is now a `max-height` on the `Form` and the
`sticky` `position` on the footer is unset when they are rendered in a `Modal`.

![dfs-update](https://user-images.githubusercontent.com/44157880/186432859-3c930ff2-db92-448d-919b-9f6c154d8a75.gif)

### Current behaviour


No custom hook to detect Form and stickyFooter use.
Footer is sticky when used in Modals, overflow is on the Modal

![dfs-borked](https://user-images.githubusercontent.com/44157880/186434983-cdd9b77b-2238-410a-af3b-95a9f2bb7b99.gif)

### Checklist




#### QA


### Additional context



### Testing instructions


See sandbox built from linked issue (you may need to add the CarbonProvider to get styling)

Worth testing in Edge (an old version if possible) as I have removed some code targeting IE10+ browsers, the code was broken anyway but will be good to confirm there isn't an issue that needs fixing

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
